hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/TypedTable.java

```diff
@@ -72,9 +73,9 @@
   public TypedTable(
       Table<byte[], byte[]> rawTable,
       CodecRegistry codecRegistry, Class<KEY> keyType,
-      Class<VALUE> valueType) throws IOException {
+      Class<VALUE> valueType, boolean statsEnabled) throws IOException {
```

Comment:
Have you considered other ways to avoid passing statsEnabled all the way through these methods? I'm not sure there is an easier solution, but maybe creating multiple methods instead of passing the boolean would be clearer.
Another thing to consider is it's not clear now (at least to me) when the statsEnabled should be true or false.

Reply:
Passing parameters is always the most explicit way of communication between software components. I think what's missing here is a neat way of passing them like a Builder or Parameter object.

I think only the tables needing exposing metrics should have stats enabled initially. And that would avoid other tables from paying the cost of keeping the counters (mainly on the locks of `AutomicLong`.
On second thought, that cost may be small and eventually, we'll enable metrics for all tables. So stats should be always there and that leaves the Table interface intact.